File: shop-admin-backend/src/main/java/com/demo/admin/modules/shop/models/ShopModel.java
package com.demo.admin.modules.shop.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.sql.Timestamp;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "shops")
public class ShopModel {
    @Id
    @Column(name = "id")
    private Long id;

    @Column(name = "user_id")
    private Long userId;

    @Column(name = "item_count")
    private Integer itemCount;

    @Column(name = "name", length = 255)
    private String name;

    @Column(name = "cover", length = 255)
    private String cover;

    @Column(name = "follower_count")
    private Integer followerCount;

    @Column(name = "rating_star")
    private Integer ratingStar;

    @Column(name = "rating_bad")
    private Integer ratingBad;

    @Column(name = "rating_good")
    private Integer ratingGood;

    @Column(name = "rating_normal")
    private Integer ratingNormal;

    @Column(name = "status")
    private Integer status;

    @Column(name = "shop_location", length = 255)
    private String shopLocation;

    @Column(name = "username", length = 255)
    private String username;

    @Column(name = "portrait", length = 255)
    private String portrait;

    @Column(name = "response_rate")
    private Integer responseRate;

    @Column(name = "country", length = 255)
    private String country;

    @Column(name = "response_time")
    private Integer responseTime;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "followed")
    private Boolean followed;

    @CreationTimestamp
    @Column(name = "last_active_time", updatable = false, insertable = false)
    private Timestamp lastActiveTime;

    @Column(name = "is_official_shop")
    private Boolean isOfficialShop;

    @Column(name = "is_active")
    private Boolean isActive;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false, insertable = false)
    private Timestamp createdAt;

    @CreationTimestamp
    @Column(name = "updated_at", updatable = false, insertable = false)
    private Timestamp updatedAt;

}

File: shop-frontend/src/models/shop.model.ts
import { BaseModel, IBaseModel } from '@core/interfaces'
import { Expose } from 'class-transformer'

export interface IShopModel extends IBaseModel {
	id: number
	user_id: number
	item_count: number
	name: string
	cover: string
	follower_count: number
	rating_star: number
	rating_bad: number
	rating_good: number
	rating_normal: number
	status: number
	shop_location: string
	username: string
	portrait: string
	response_rate: number
	country: string
	response_time: number
	description: string
	followed: number
	last_active_time: string
	is_official_shop: number
	created_at: Date
	updated_at: Date | null
}

export class ShopModel extends BaseModel implements IShopModel {
	@Expose()
	id!: number

	@Expose()
	user_id!: number

	@Expose()
	item_count!: number

	@Expose()
	name!: string

	@Expose()
	cover!: string

	@Expose()
	follower_count!: number

	@Expose()
	rating_star!: number

	@Expose()
	rating_bad!: number

	@Expose()
	rating_good!: number

	@Expose()
	rating_normal!: number

	@Expose()
	status!: number

	@Expose()
	shop_location!: string

	@Expose()
	username!: string

	@Expose()
	portrait!: string

	@Expose()
	response_rate!: number

	@Expose()
	country!: string

	@Expose()
	response_time!: number

	@Expose()
	description!: string

	@Expose()
	followed!: number

	@Expose()
	last_active_time!: string

	@Expose()
	is_official_shop!: number

	@Expose()
	created_at!: Date

	@Expose()
	updated_at!: Date | null
}
export class ShopBaseModel extends BaseModel {
	@Expose()
	id!: number

	@Expose()
	name!: string

	@Expose()
	username!: string

	@Expose()
	portrait!: string

	@Expose()
	last_active_time!: string

	@Expose()
	is_official_shop!: number
}
